### DISCUSSION TITLE
Triggering Airflow DAGs based on events

### DISCUSSION THREAD

**Discussion by aniquedavla** [apache/airflow#26832] [Q&A] [answered]
Airflow docs assert that Airflow was designed for scheduled tasks that have defined start and end times, aka don't run infinitely. However Airflow provides a Rest Api to trigger DAGs and to trigger them from CLI.

Can someone clarify if there are limitations or problems with using Airflow to run DAGs that would be triggered via the API? Let's say if I want to run couple hundred to thousand instances of a DAG a day, each triggered by some event? The workflows would not run infinitely.

**Comment by potiuk** [comment, marked as answer]
Likely there are limitations. Very much depending on your deployment/resources, how heavy your tasks are, which executors you have etc. etc. And you neeed to test it yourself. There is no HARD stop at specific numbers. You just need to assume that for every task started there are overheads regarding Python interpreter spawning, Kubernetes code spawning tasks passing over quees, tasks serializing over multiprocessing (all depends on the type of executor you have).

Generally if you have many very short tasks (basically trying to emulate streaming), you will get a lot of overhead for just managing the tasks because airflow is not designed to handle very small tasks efficiently - usually a task in airflow is coarse-grained rather than very fine-grained/tiny micro-task. It's just very costly to use airflow as a streaming solution.

**Reply by aniquedavla** [reply]
thank you so much for clarifying that @potiuk
